src/freenas/usr/local/lib/middlewared_truenas/plugins/failover.py

```diff
@@ -1479,19 +1479,33 @@ async def hook_restart_devd(middleware, *args, **kwargs):
 
 async def hook_license_update(middleware, *args, **kwargs):
     FailoverService.HA_MODE = None
-    if await middleware.call('failover.licensed'):
-        await middleware.call('failover.ensure_remote_client')
-        etc_generate = ['rc']
-        if await middleware.call('system.feature_enabled', 'FIBRECHANNEL'):
-            await middleware.call('etc.generate', 'loader')
-            etc_generate += ['loader']
-        try:
-            await middleware.call('failover.send_small_file', '/data/license')
-            await middleware.call('failover.call_remote', 'failover.ensure_remote_client')
-            for etc in etc_generate:
-                await middleware.call('failover.call_remote', 'etc.generate', [etc])
-        except Exception:
-            middleware.logger.warning('Failed to sync license file to standby.')
+
+    if not await middleware.call('failover.licensed'):
+        return
+
+    # setup the local controller first
+    await middleware.call('failover.ensure_remote_client')
+    etc_generate = ['rc']
+    if await middleware.call('system.feature_enabled', 'FIBRECHANNEL'):
+        await middleware.call('etc.generate', 'loader')
+        etc_generate += ['loader']
+
+    # setup the remote controller
+    try:
+        await middleware.call('failover.send_small_file', '/data/license')
+    except Exception:
+        middleware.logger.warning('Failed to sync db to standby')
+
+    try:
+        await middleware.call('failover.call_remote', 'failover.ensure_remote_client')
+    except Exception:
+        middleware.logger.warning('Failed to ensure remote client on standby')
+
+    try:
```

Comment:
The way they were placed wasn't a mistake. If `failover.ensure_remote_client` there is no point in trying to call failover.call_remote next